parquet-avro/src/main/java/org/apache/parquet/avro/AvroSchemaConverter.java

```diff
@@ -150,16 +151,23 @@ public MessageType convert(Schema avroSchema) {
     if (!avroSchema.getType().equals(Schema.Type.RECORD)) {
       throw new IllegalArgumentException("Avro schema must be a record.");
     }
-    return new MessageType(avroSchema.getFullName(), convertFields(avroSchema.getFields(), ""));
+    return new MessageType(
+        avroSchema.getFullName(),
+        convertFields(avroSchema.getFields(), "", new IdentityHashMap<Schema, Integer>()));
```

Comment:
You don't need a map, only a set since you only add the same element once. (Maybe use `HashSet`?)

Isn't it enough to use the type name only instead of the whole `Schema` object?

Reply:
I'm sorry I was wrong. Your choice of using `IdentityHashMap` was the best approach. (Unfortunately, there is no such thing in standard Java as `IdentityHashSet`.)

Could you change it back using `IdentityHashMap`? (Maybe with the value type `Void` and setting `null` values to be clear we don't actually use them.)